src/screens/game-details-modal/components/lol-games.tsx

```diff
@@ -33,6 +48,16 @@ const LolGame = ({ number, draft, duration, score }: LolGameProps) => {
     return `${paddedMinutes}:${paddedSeconds}`;
   }
 
+  const { replayVideo, openReplayVideo, searchReplay } = useReplay();
+  useEffect(() => {
+    searchReplay({
+      date,
+      teams,
+      game: competitionName,
+      ...(gameNumbers > 1 ? { gameNumber: number } : {}),
+    });
+  }, [replayVideo, openReplayVideo, searchReplay]);
```

Comment:
You don't need to add `replayVideo` and `openReplayVideo` to dependencies (it could reduce performance). But I think you need to add `date`, `teams`, `competionName` and `gameNumbers`.